homeassistant/components/ecobee/binary_sensor.py

```diff
@@ -99,11 +99,16 @@ def device_class(self):
     async def async_update(self):
         """Get the latest state of the sensor."""
         await self.data.update()
-        for sensor in self.data.ecobee.get_remote_sensors(self.index):
-            if sensor["name"] != self.sensor_name:
-                continue
-            for item in sensor["capability"]:
-                if item["type"] != "occupancy":
+        thermostat = self.data.ecobee.get_thermostat(self.index)
+        if not thermostat["runtime"]["connected"]:
+            # If the thermostat is disconnected, sensor data is stale
+            self._state = "false"
```

Comment:
Do we know that this is the true state? Maybe set `available` property to False instead?

Reply:
I've replaced this with the `available` property.